File: backend/contracts/agent_lane_order.py
"""Per-lane agent ordering for dashboard persistence and the workflow graph UI.

Ticker runs created with ``entry_path="profiler"`` use ``PROFILER_ENTRY_AGENT_NAMES``.
Surveyor-discovered lanes use ``SURVEYOR_ENTRY_AGENT_NAMES`` (no profiler execution).

The graph column order for profiler-entry lanes matches ``PROFILER_ENTRY_AGENT_NAMES``;
the dashboard frontend mirrors these tuples in ``frontend/src/graph/agentLaneOrder.ts``.
``LANE_AGENT_SLUGS`` is the superset used when filtering ticker-lane agent rows.
``tests/backend/unit/test_agent_lane_order_sync.py`` asserts both stay aligned.
"""

from typing import Final

PROFILER_ENTRY_AGENT_NAMES: Final[tuple[str, ...]] = (
    "profiler",
    "researcher",
    "strategist",
    "sentinel",
    "appraiser",
)

SURVEYOR_ENTRY_AGENT_NAMES: Final[tuple[str, ...]] = (
    "researcher",
    "strategist",
    "sentinel",
    "appraiser",
)

# Superset of surveyor-entry lane agents; profiler is absent on surveyor-entry runs.
LANE_AGENT_SLUGS: Final[frozenset[str]] = frozenset(PROFILER_ENTRY_AGENT_NAMES)

File: frontend/src/graph/agentLaneOrder.ts
import type { AgentNameSlug } from "../api";

/** Matches ``backend.contracts.agent_lane_order.PROFILER_ENTRY_AGENT_NAMES``. */
export const PROFILER_ENTRY_AGENT_NAMES = [
  "profiler",
  "researcher",
  "strategist",
  "sentinel",
  "appraiser",
] as const satisfies readonly AgentNameSlug[];

/** Matches ``backend.contracts.agent_lane_order.SURVEYOR_ENTRY_AGENT_NAMES``. */
export const SURVEYOR_ENTRY_AGENT_NAMES = [
  "researcher",
  "strategist",
  "sentinel",
  "appraiser",
] as const satisfies readonly AgentNameSlug[];

/** Agent slugs that may appear on any ticker lane (profiler- or surveyor-entry). */
export const LANE_AGENT_SLUGS = new Set<string>(PROFILER_ENTRY_AGENT_NAMES);

/** Column / chain ordering for profiler-entry lanes in ``buildGraphLayout``. */
export const GRAPH_LAYOUT_PROFILER_LANE_ORDER: readonly AgentNameSlug[] =
  PROFILER_ENTRY_AGENT_NAMES;
